custom_components/frigate/camera.py

```diff
@@ -67,25 +74,46 @@ async def async_setup_entry(
     client_id = get_frigate_instance_id_for_config_entry(hass, entry)
     coordinator = hass.data[DOMAIN][entry.entry_id][ATTR_COORDINATOR]
 
-    async_add_entities(
-        [
-            FrigateCamera(
-                entry,
-                cam_name,
-                frigate_client,
-                client_id,
-                coordinator,
-                frigate_config,
-                camera_config,
+    if entry.options.get(CONF_ENABLE_WEBRTC, False):
```

Comment:
Maybe you can reduce the size and improve the readability here with something like:

```py
frigate_webrtc = entry.options.get(CONF_ENABLE_WEBRTC, False)
camera_type = FrigateCameraWebRTC if  frigate_webrtc else FrigateCamera
birdseye_type = BirdseyeCameraWebRTC if frigate_webrtc else BirdseyeCamera

async_add_entities([
  camera_type(...) + (birdseye_type() if ...)
```

Might be a bit cleaner and force us to keep the constructor arguments the same?

Reply:
Awesome tip! My Python knowledge is rudimentary :P

Reply:
On the code duplication topic, it should not be too hard to extract the duplicated stuff from `FrigateCamera` and `BirdseyeCamera` classes into a `FrigateStreamCamera` class that both can inherit from.

Maybe I'll do this someday.